### ISSUE TITLE
New Recurring Transaction switching Type of Transaction works partly not as expected

### ISSUE THREAD

**Issue by wuk986** [moneymanagerex/moneymanagerex#3815]
MMEX version:
 - [x] 1.5.x
 - [ ] 1.6.x

Note: bug reporters are expected to have verified the bug still exists
either in the last stable version of MMEX or on updated development code
(master branch).

Operating System:
 - [x] Windows
 - [x] Mac OSX
 - [ ] Linux 

# Description of the bug

When you create a new recurring transaction and change the type from _Withdrawal_ to _Transfer_ and back the dialog window Select account is still present instead of Organize Payees

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:

1. Open the New Recurring Transaction window
2. Switch the Type from Withdrawal to Transfer
3. Click Select to Account
4. Switch the Type from Transfer to Withdrawal or Deposit
5. Click Select Account

…

Expected result:

When I switch back from Transfer to Withdrawal or Deposit I expect to get the window Organize Payees.

Actual result:

When I switch back from Transfer to Withdrawal or Deposit I get always the window Select Account.

# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by tactilis** [comment]
I can reproduce this but I think you have made a small error in the wording of Step 5 above.

The reproduction steps are:

1. Open the New Recurring Transaction window
2. Switch the Type from Withdrawal to Transfer
3. Click 'Select To Account'  and choose an account.
4. Switch the Type from Transfer to Withdrawal or Deposit. Note that the button clicked at Step 3 now correctly says 'Select Payee'
5. Click **'Select Payee'**

At this point the Select Account dialog is shown, whereas the Organize Payees dialog should be displayed.

**Comment by whalley** [comment]
Yes, I can reproduce. I think it is also worth tidying this up and making the dialog similar to the normal transaction dialog and having the account and to account/payee as combo boxes.

Recurring
![image](https://user-images.githubusercontent.com/487345/130744479-ce711ca8-3aed-44f1-845c-05963eba248d.png)

Account Transaction
![image](https://user-images.githubusercontent.com/487345/130745051-1052f41d-00c0-4171-8a35-def790096ee5.png)

**Comment by whalley** [comment]
Bug fixed and new issue raised for enhancement.
